cdap-app-fabric/src/test/java/co/cask/cdap/app/runtime/monitor/RuntimeMonitorTest.java

```diff
@@ -153,52 +172,161 @@ public void stop() {
+    RuntimeMonitor runtimeMonitor = new RuntimeMonitor(programRunId, monitorCConf, monitorClient,
+                                                       datasetFramework, transactional, messagingContext, scheduler);
 
     runtimeMonitor.startAndWait();
     // use different configuration for verification
-    String lastProcessed = verifyPublishedMessages(2, cConfCopy, 2, null);
+    String lastProcessed = verifyPublishedMessages(monitorCConf, 2, null);
     runtimeMonitor.stopAndWait();
 
     // publish some more messages to test offset manager
-    publishProgramStatus(programRunId);
-    verifyPublishedMessages(2, cConf, 2, lastProcessed);
+    publishProgramStatus(programRunId, ProgramStatus.RUNNING);
+    publishProgramStatus(programRunId, ProgramStatus.RUNNING);
+    verifyPublishedMessages(cConf, 2, lastProcessed);
 
-    runtimeMonitor = new RuntimeMonitor(programRunId, cConfCopy, messagingService, monitorClient,
-                                        datasetFramework, transactionSystemClient);
+    runtimeMonitor = new RuntimeMonitor(programRunId, monitorCConf, monitorClient,
+                                        datasetFramework, transactional, messagingContext, scheduler);
     runtimeMonitor.startAndWait();
     // use different configuration for verification
-    lastProcessed = verifyPublishedMessages(2, cConfCopy, 2, lastProcessed);
+    lastProcessed = verifyPublishedMessages(monitorCConf, 2, lastProcessed);
 
     // publish completed status to trigger offset clean up
-    publishCompletedStatus(programRunId);
+    publishProgramStatus(programRunId, ProgramStatus.COMPLETED);
 
     // use different configuration for verification
-    verifyPublishedMessages(2, cConfCopy, 1, lastProcessed);
+    verifyPublishedMessages(monitorCConf, 1, lastProcessed);
 
-    // wait for runtime server to stop automatically
-    Tasks.waitFor(true, () -> !runtimeServer.isRunning(), 5, TimeUnit.MINUTES);
+    // The RuntimeServer should be stopped upon the RuntimeMonitor received the program completed state
+    Tasks.waitFor(Service.State.TERMINATED, runtimeServer::state, 10, TimeUnit.SECONDS);
+    // The RuntimeMonitor should stop itself upon receiving the program completed state
+    Tasks.waitFor(Service.State.TERMINATED, runtimeMonitor::state, 10, TimeUnit.SECONDS);
+  }
 
-    runtimeMonitor.stopAndWait();
+  @Test
+  public void testTopicExpansion() throws Exception {
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(5, Threads.createDaemonThreadFactory("test"));
+
+    RunId runId = RunIds.generate();
+    ProgramRunId programRunId = NamespaceId.DEFAULT.app("app1").workflow("testTopicExpansion").run(runId);
+
+    // change topic name because cdap config is different than runtime config
+    CConfiguration monitorCConf = CConfiguration.copy(cConf);
+
+    monitorCConf.set(Constants.AppFabric.PROGRAM_STATUS_EVENT_TOPIC, "cdap-programStatus");
```

Comment:
we are setting the topic name to same as default no?

Reply:
cdap-programStatus is not the default one, right?